editor/editor_themes.cpp

```diff
@@ -1911,45 +1917,43 @@ Ref<Theme> create_editor_theme(const Ref<Theme> p_theme) {
+	theme->set_stylebox("panel", "GraphElement", graphn_sb_panel);
+	theme->set_stylebox("panel_selected", "GraphElement", graphn_sb_panel_selected);
+	theme->set_stylebox("titlebar", "GraphElement", graphn_sb_titlebar);
+	theme->set_stylebox("titlebar_selected", "GraphElement", graphn_sb_titlebar_selected);
+
+	// GraphNode's title Label
+	theme->set_type_variation("GraphNodeTitleLabel", "Label");
+
+	theme->set_stylebox("normal", "GraphNodeTitleLabel", memnew(StyleBoxEmpty));
+	theme->set_color("font_color", "GraphNodeTitleLabel", font_color);
+	theme->set_constant("line_spacing", "GraphNodeTitleLabel", 3 * EDSCALE);
+
+	Color graphnode_decoration_color = dark_color_1.inverted();
+
+	theme->set_color("resizer_color", "GraphElement", graphnode_decoration_color);
+	theme->set_icon("resizer", "GraphElement", theme->get_icon(SNAME("GuiResizer"), SNAME("EditorIcons")));
 
-	Color node_decoration_color = dark_color_1.inverted();
-	theme->set_color("title_color", "GraphNode", node_decoration_color);
-	node_decoration_color.a = 0.7;
-	theme->set_color("close_color", "GraphNode", node_decoration_color);
-	theme->set_color("resizer_color", "GraphNode", node_decoration_color);
-
-	theme->set_constant("port_offset", "GraphNode", 0);
-	theme->set_constant("title_h_offset", "GraphNode", 12 * EDSCALE);
-	theme->set_constant("title_offset", "GraphNode", 21 * EDSCALE);
-	theme->set_constant("close_h_offset", "GraphNode", -2 * EDSCALE);
-	theme->set_constant("close_offset", "GraphNode", 20 * EDSCALE);
+	// GraphNode
+	theme->set_stylebox("panel", "GraphNode", graphn_sb_panel);
+	theme->set_stylebox("panel_selected", "GraphNode", graphn_sb_panel_selected);
+	theme->set_stylebox("titlebar", "GraphNode", graphn_sb_titlebar);
+	theme->set_stylebox("titlebar_selected", "GraphNode", graphn_sb_titlebar_selected);
+	theme->set_stylebox("slot", "GraphNode", graphn_sb_slot);
+
+	theme->set_color("resizer_color", "GraphNode", graphnode_decoration_color);
+
+	theme->set_constant("port_h_offset", "GraphNode", 0);
 	theme->set_constant("separation", "GraphNode", 1 * EDSCALE);
 
-	theme->set_icon("close", "GraphNode", theme->get_icon(SNAME("GuiCloseCustomizable"), SNAME("EditorIcons")));
-	theme->set_icon("resizer", "GraphNode", theme->get_icon(SNAME("GuiResizer"), SNAME("EditorIcons")));
 	Ref<ImageTexture> port_icon = theme->get_icon(SNAME("GuiGraphNodePort"), SNAME("EditorIcons"));
 	// The true size is 24x24 This is necessary for sharp port icons at high zoom levels in GraphEdit (up to ~200%).
 	port_icon->set_size_override(Size2(12, 12));
 	theme->set_icon("port", "GraphNode", port_icon);
 
-	theme->set_font("title_font", "GraphNode", theme->get_font(SNAME("main_bold_msdf"), SNAME("EditorFonts")));
+	//TODO: @Geometror Change to theme variations.
+	theme->set_stylebox("state_machine_frame", "GraphNode", smgraphsb);
+	theme->set_stylebox("state_machine_selected_frame", "GraphNode", smgraphsbselected);
```

Comment:
Do you want to do this now or in a follow-up?

Reply:
I looked at the actual usage in the animation tree editor. They aren't used as a replacement/variation of the GraphNode's "body" stylebox, but for some kin of overlay. However, after observing the AnimationTree in our platformer demo I'm not quite sure whether this overlay is actually drawn at the moment. For now, I removed the comment.